internal/server/singleprocess/service_job_test.go

```diff
@@ -794,3 +790,343 @@ func TestServiceQueueJob_odr_default(t *testing.T) {
 		require.NotNil(open)
 	}
 }
+
+func TestServiceQueueJob_odr_target_id(t *testing.T) {
+	require := require.New(t)
+	ctx := context.Background()
+
+	log := hclog.New(&hclog.LoggerOptions{
+		Name:  "odr-test",
+		Level: hclog.Trace,
+	})
+
+	ctx = hclog.WithContext(ctx, log)
+
+	// Create our server
+	impl, err := New(
+		WithLogger(log),
+		WithDB(testDB(t)),
+	)
+
+	require.NoError(err)
+	client := server.TestServer(t, impl, server.TestWithContext(ctx))
+
+	// Initialize our app
+	TestApp(t, client, serverptypes.TestJobNew(t, &pb.Job{
+		Application: &pb.Ref_Application{
+			Application: "app",
+			Project:     "proj",
+		},
+	}).Application)
+
+	// Simplify writing tests
+	type Req = pb.QueueJobRequest
+
+	// Create an ODR profile with target runner ID
+	runnerId := "test_r"
+	odr := serverptypes.TestOnDemandRunnerConfig(t, &pb.OnDemandRunnerConfig{
+		Name:         "test",
+		PluginType:   "magic-carpet",
+		PluginConfig: []byte("foo = 1"),
+		EnvironmentVariables: map[string]string{
+			"CARPET_DRIVER": "apu",
```

Comment:
![](https://static.wikia.nocookie.net/thedescendants/images/1/17/Aladdin_Disney_Apu_and_the_Magic_Carpet.jpg/revision/latest/scale-to-width-down/250?cb=20160613003309)